misc/gen_types.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
+
+
+import argparse
+from pathlib import Path
+import json
+import re
+
+FIELDS_TEMPLATE = """
+    @property
+    def {name}(self) -> {type}: ..."""
+
+
+def replace_type(typename):
+    REGEX_REPLACEMENTS = [
+        (r"Option<(.*)>", "Optional[{type}]"),
+        (r"List<(.*)>", "List[{type}]"),
+    ]
+
+    KEYWORD_REPLACEMENT = [("Bytes", "bytes"), ("Float", "float")]
+
+    for (reg, format) in REGEX_REPLACEMENTS:
+        match = re.match(reg, typename)
+
+        if match:
+            value = match.group(1)
+            return format.format(type=value)
```

Comment:
If I have `typename` to `Option<Bytes>` this would return `Optional[Bytes]`, right ?

Reply:
Yes, the script tries to translate `rust` types to python ones. I've added Option<T> and List since it is the only types I've seen during the migration.

Reply:
But `Bytes` should be translated to `bytes`, right ?

What I'm pointing is that your forgot to translate the wrapped type by the `Option`